File: blog-sever/server-java/src/main/java/com/site/blog/util/ResultGenerator.java
package com.site.blog.util;

import com.site.blog.constants.HttpStatusEnum;
import com.site.blog.model.dto.Result;


public class ResultGenerator {

    private ResultGenerator() {
    }

    public static <T> Result<T> getResultByHttp(HttpStatusEnum constants, String msg, T data) {
        Result<T> result = new Result<>();
        result.setResultCode(constants.getStatus());
        result.setMessage(msg);
        result.setData(data);
        result.setTimestamp(System.currentTimeMillis());
        return result;
    }

    public static <T> Result<T> getResultByHttp(HttpStatusEnum constants, T data) {
        Result<T> result = new Result<>();
        result.setResultCode(constants.getStatus());
        result.setMessage(constants.getContent());
        result.setData(data);
        result.setTimestamp(System.currentTimeMillis());
        return result;
    }

    public static <T> Result<T> getResultByHttp(HttpStatusEnum constants, boolean success, T data) {
        Result<T> result = new Result<>();
        result.setResultCode(constants.getStatus());
        result.setMessage(constants.getContent());
        result.setData(data);
        result.setSuccess(success);
        result.setTimestamp(System.currentTimeMillis());
        return result;
    }

    /**
     * 自定义提示消息
     *
     * @param constants Http枚举
     * @param msg       提示消息
     */
    public static Result<String> getResultByMsg(HttpStatusEnum constants, String msg) {
        Result<String> result = new Result<>();
        result.setResultCode(constants.getStatus());
        result.setMessage(msg);
        result.setTimestamp(System.currentTimeMillis());
        return result;
    }


    public static Result<String> getResultByHttp(HttpStatusEnum constants) {
        Result<String> result = new Result<>();
        result.setResultCode(constants.getStatus());
        result.setMessage(constants.getContent());
        result.setTimestamp(System.currentTimeMillis());
        return result;
    }
}

File: myblog-kt/src/main/kotlin/com/site/blog/util/ResultGenerator.kt
package com.site.blog.util

import com.site.blog.constants.HttpStatusEnum
import com.site.blog.model.dto.Result

object ResultGenerator {
    @JvmStatic
    fun <T> getResultByHttp(constants: HttpStatusEnum, msg: String?, data: T): Result<T> {
        val result = Result<T>()
        result.resultCode = constants.status
        result.message = msg
        result.data=data
        result.timestamp = System.currentTimeMillis()
        return result
    }
    @JvmStatic
    fun <T> getResultByHttp(constants: HttpStatusEnum, data: T): Result<T> {
        val result = Result<T>()
        result.resultCode = constants.status
        result.message = constants.content
        result.data=data
        result.timestamp = System.currentTimeMillis()
        return result
    }
@JvmStatic
    fun <T> getResultByHttp(constants: HttpStatusEnum, success: Boolean, data: T): Result<T> {
        val result = Result<T>()
        result.resultCode = constants.status
        result.message = constants.content
        result.data=data
        result.success = success
        result.timestamp = System.currentTimeMillis()
        return result
    }

    /**
     * 自定义提示消息
     *
     * @param constants Http枚举
     * @param msg       提示消息
     */
    @JvmStatic
    fun getResultByMsg(constants: HttpStatusEnum, msg: String?): Result<String> {
        val result = Result<String>()
        result.resultCode = constants.status
        result.message = msg
        result.timestamp = System.currentTimeMillis()
        return result
    }

    @JvmStatic
    fun getResultByHttp(constants: HttpStatusEnum): Result<String> {
        val result = Result<String>()
        result.resultCode = constants.status
        result.message = constants.content
        result.timestamp = System.currentTimeMillis()
        return result
    }
}